src/screens/NewsDetailsScreen.js

```diff
@@ -53,3 +35,11 @@ const styles = StyleSheet.create({
     flexDirection: 'row',
   },
 });
+
+const mapStateToProps = state => {
```

Comment:
use conventional function, also you need to use second argument `ownProps` to access navigation...url prop

```
function mapStateToProps(state, ownProps) {
  const { url } = navigation.state.params;
   //here you need to implement selector getArticle that
  //knows how to access state.articles and find article based on url
  const article = getArticle(state, url); 
}
```